### PR TITLE
add rubberband select to list view mode

### PR DESCRIPTION
**Pull request by lukefromdc**
Based on https://github.com/mate-desktop/caja/pull/1062 but with a build warning fixed and a possible memory leak freed

### CODE REVIEW

**Comment by lukefromdc** [comment]
Note that there is one behavioral change, in that drag and drop is possible ONLY from the icon and not the entire tree row with this change

**Comment by lukefromdc** [comment]
I've installed a "dev-all" build of Caja with both this and https://github.com/mate-desktop/caja/pull/1067 applied so I can watch this in terms of behavior

File: src/file-manager/fm-list-view.c [hunk 1]
```diff
@@ -652,20 +683,137 @@ do_popup_menu (GtkWidget *widget, FMListView *view, GdkEventButton *event)
     }
 }
 
+
```
Comment (raveit65) [inline src/file-manager/fm-list-view.c]: Do we need those extra lines?

File: src/file-manager/fm-list-view.c [hunk 2]
```diff
@@ -652,20 +683,137 @@ do_popup_menu (GtkWidget *widget, FMListView *view, GdkEventButton *event)
...
+    /* determine column/row area occupied by cell renderers */
+    for (l = cell_renderers; l != NULL; l = l->next) {
+        gtk_tree_view_get_cell_area (tree_view, path, column, &rect);
+        gtk_tree_view_column_cell_get_position (column, GTK_CELL_RENDERER (l->data),
+                                &x_offset, &width);
+        rect.x += x_offset;
+        rect.width = width;
+
+        gtk_cell_renderer_get_size (GTK_CELL_RENDERER (l->data),
```
Comment (raveit65) [inline src/file-manager/fm-list-view.c]: this gives me a deprecation warning.
```
fm-list-view.c: In function 'cell_event_hit_test':
fm-list-view.c:749:9: warning: 'gtk_cell_renderer_get_size' is deprecated: Use 'gtk_cell_renderer_get_preferred_size' instead [-Wdeprecated-declarations]
         gtk_cell_renderer_get_size (GTK_CELL_RENDERER (l->data),
         ^~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /usr/include/gtk-3.0/gtk/gtktreeviewcolumn.h:25,
                 from /usr/include/gtk-3.0/gtk/gtktreeview.h:27,
                 from /usr/include/gtk-3.0/gtk/gtkcombobox.h:27,
                 from /usr/include/gtk-3.0/gtk/gtkappchooserbutton.h:29,
                 from /usr/include/gtk-3.0/gtk/gtk.h:42,
                 from fm-directory-view.h:31,
                 from fm-list-view.h:30,
                 from fm-list-view.c:29:
/usr/include/gtk-3.0/gtk/gtkcellrenderer.h:225:18: note: declared here
 void             gtk_cell_renderer_get_size       (GtkCellRenderer      *cell,
                  ^~~~~~~~~~~~~~~~~~~~~~~~~~
```

**Comment by raveit65** [comment]
Hmm, open a file or folder in list-view with double-click doesn't work any more with that commit.
Can you confirm?

**Comment by lukefromdc** [comment]
You have to double-click the icon itself (first column) to open the file. Between that and the deprecation warning (too complex for me to fix right now and not my code), we might want to close this. The behavior change on double-clicking files is likely to confuse people, and is apparently necessary for this to work, as the rubberbanding code had to exclude the first column,

**Comment by raveit65** [comment]
Ok, that is working.
Personal i don't use the list-view, but is that the expected behaviour for using the list-view?
If you you don't know that you will think it is a bug, same like me.
So, let's close this.

**Comment by raveit65** [comment]
Ok, that is working.
Personal i don't use the list-view, but is that the expected behaviour for using the list-view?
If you you don't know that you will think it is a bug, same like me.
So, let's close this.

**Comment by raveit65** [comment]
Ok, that is working.
Personal i don't use the list-view, but is that the expected behaviour for using the list-view?
If you you don't know that you will think it is a bug, same like me.
So, let's close this.

**Pull request closed without merging**